batcher/batcher.go

```diff
@@ -16,9 +16,11 @@ type Batcher struct {
 	timeout   time.Duration
 	prefilter func(interface{}) error
 
-	lock   sync.Mutex
-	submit chan *work
-	doWork func([]interface{}) error
+	lock              sync.Mutex
+	submit            chan *work
+	doWork            func([]interface{}) error
+	workCount         int
+	callbackWorkAdded func(count int)
```

Comment:
The callback and workCount seem to only be used for testing, and I would prefer not to pollute the production code for the sake of tests, especially when the change is otherwise so trivial.

Reply:
I updated the code. I removed the extra fields workCount and callbackWorkAdded. I added sleep in TestFlushSuccess. @eapache please, check it again.

Reply:
renamed the method to Shutdown and added parameter wait bool